src/context-selection/attribute-option-combo-selector-bar-item/use-selected.js

```diff
@@ -2,14 +2,39 @@ import { useCallback } from 'react'
 import { useAttributeOptionComboSelection } from '../use-context-selection/index.js'
 import omit from './omit.js'
 
-export default function useSelected() {
+export default function useSelected(relevantCategoriesWithOptions) {
     const [attributeOptionComboSelection, setAttributeOptionComboSelection] =
         useAttributeOptionComboSelection()
 
-    const deselectAll = useCallback(
-        () => setAttributeOptionComboSelection(undefined),
-        [setAttributeOptionComboSelection]
-    )
+    const deselectAll = useCallback(() => {
+        const nextAocSelection = Object.fromEntries(
+            Object.entries(attributeOptionComboSelection).filter(
+                ([categoryId, optionId]) => {
+                    return relevantCategoriesWithOptions.find(
+                        (category) =>
+                            category.id === categoryId &&
+                            category.categoryOptions.find(
+                                (categoryOption) =>
+                                    categoryOption.id === optionId
+                            )
+                    )
+                }
+            )
+        )
+
+        if (
+            JSON.stringify(nextAocSelection) ===
+            JSON.stringify(attributeOptionComboSelection)
+        ) {
+            return
```

Comment:
Is this needed? What happens if you just set the same selection. Is something refetched or re-rendered? I would think setting the same selection wouldn't really do anything?

Reply:
You're right, also with the comment above, I could change the two return statements to:

```js
        if (
            JSON.stringify(nextAocSelection) !==
            JSON.stringify(attributeOptionComboSelection)
        ) {
            setAttributeOptionComboSelection(nextAocSelection)
        }
```

Reply:
You're right. I can completely omit the condition and the two return statements (which also addresses your comment above)